### ISSUE TITLE
Availability statuses

### ISSUE THREAD

**Issue by uber42** [lni/dragonboat#235]
Is it planned to introduce statuses for cluster nodes?
For example
- "Available"
- "Unavailable", when the heartbeat cannot be sent for some timeout
- "Stale", when follower is far behind the leader (commitedIndex maybe)

**Comment by lni** [comment]
There were some similar discussions on whether to provide such availability info. My concern is that such availability info determined by a raft library might not a 100% match with what users really want, e.g. dragonboat considers a remote node to be "unavailable" after some timeouts, it tells this to a program totally out of the control of dragonboat, that program starts to believe that the remote node is definitely "unavailable" and start making decisions based on that, then eventually there could be problems when the remote node is actual available. 

In short, I don't think a consensus library should tell users whether their nodes are available or not, a consensus library is just a consensus library, especially when there is no way to reliably tell whether a node is really down/unavailable. 

For the "stale" status, that is something we can discuss, as it is something that should be reported by the consensus library.
